src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -869,7 +869,7 @@ public string[] Segments
                     }
                     else
                     {
-                        LowLevelList<string> pathSegments = new LowLevelList<string>();
+                        var pathSegments = new ArrayBuilder<string>();
```

Comment:
I'm deciding to change the allocation pattern back to 0, 4, 8, etc. so we don't have to test this every time we make such a change. @jkotas While it may be true that 1-length arrays are very common, having the initial capacity at 1 will penalize for any non-1 length.